File: CSharp/PaymentsService/Infrastructure/Payments/CreditCard/CreditCardPaymentFactory.cs
using PaymentsService.Domain.Payments;

namespace PaymentsService.Infrastructure.Payments.CreditCard;
public sealed class CreditCardPaymentFactory : IPaymentAbstractFactory
{
    public IPaymentClient GetPaymentClient()
    {
        return new CreditCardPaymentClient();
    }

    public IProcessPaymentRequestValidator GetProcessPaymentRequestValidator()
    {
        return CreditCardProcessPaymentRequestValidator.Instance;
    }

    public ProcessPaymentBaseResponse GetValidationFailedResponse(RequestValidationResult result)
    {
        return new ProcessCreditCardPaymentResponse()
        {
            Status = "ValidationFailed",
            Error = result.Error
        };
    }
}

File: Python/PaymentsService/Infrastructure/Payments/CreditCard/CreditCardPaymentFactory.py
from Domain.Payments.IPaymentAbstractFactory import IPaymentAbstractFactory
from Domain.Payments.IPaymentClient import IPaymentClient
from Domain.Payments.IProcessPaymentRequestValidator import IProcessPaymentRequestValidator
from Domain.Payments.ProcessPaymentBaseResponse import ProcessPaymentBaseResponse
from Infrastructure.Payments.CreditCard.CreditCardPaymentClient import CreditCardPaymentClient
from Infrastructure.Payments.CreditCard.CreditCardProcessPaymentRequestValidator import CreditCardProcessPaymentRequestValidator
from Infrastructure.Payments.CreditCard.ProcessCreditCardPaymentResponse import ProcessCreditCardPaymentResponse


class CreditCardPaymentFactory(IPaymentAbstractFactory):
    def get_payment_client(self) -> IPaymentClient:
        return CreditCardPaymentClient()

    def get_process_payment_request_validator(self) -> IProcessPaymentRequestValidator:
        return CreditCardProcessPaymentRequestValidator()

    def get_validation_failed_response(self, result) -> ProcessPaymentBaseResponse:
        return ProcessCreditCardPaymentResponse(status="ValidationFailed", error=result.error)